vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/SlideManager.java

```diff
@@ -46,4 +54,36 @@ public ISlide getSlide(String slideId) {
     public void updateSlide(Slide slide) {
         slideRepo.save((Slide) slide);
     }
+    
+	@Override
+	public void deleteSlideById(String slideId, String moduleId, String flag) throws SlideDoesNotExistException {
+		try {
+
+			if (flag.equals("0")) {
+				slideRepo.delete((Slide) getSlide(slideId));
+			}
+			// Slide part of another sequence
+			else {
+				List<Sequence> sequences = sequenceRepo.findSequencesForModule(moduleId);
+				int sizeOfSequences = sequences.size();
+				for (int i = 0; i < sizeOfSequences; i++) {
+					
+					if (sequences.get(i).getSlides().size() > 0) {
+						int sizeOfSlides = sequences.get(i).getSlides().size();
+						
+						for(int j = 0; j < sizeOfSlides; j++) {
+							if (sequences.get(i).getSlides().get(j).getId().equals(slideId)) {
+								sequences.get(i).getSlides().remove(j);
+								sequenceRepo.save(sequences.get(i));
+								slideRepo.delete((Slide) getSlide(slideId));
```

Comment:
why do you delete the slide here? what it it's still in another sequence? that should be the last line in thee method

Reply:
Yes, it's a part of the sequence, so I am deleting it's references in the loop itself.

Reply:
this line will be executed every time a sequence includes the slide; so if two sequences  contain the slide you call delete slide as soon as the first sequence contains the slide (which will probably raise an exception since the slide is still in another sequence).

Reply:
Yes, I got it. I will do the changes.